internals-js/src/error.ts

```diff
@@ -1,25 +1,333 @@
 import { ASTNode, GraphQLError, Source } from "graphql";
+import { SchemaRootKind } from "./definitions";
+import { assert, MapWithCachedArrays } from "./utils";
 
-export function error(
-  code: string,
-  message: string,
-  nodes?: readonly ASTNode[] | ASTNode,
-  source?: Source,
-  positions?: readonly number[],
-  path?: readonly (string | number)[],
-  originalError?: Error,
-  extensions?: { [key: string]: any },
-) {
-  return new GraphQLError(
-    message,
-    nodes,
-    source,
-    positions,
-    path,
-    originalError,
-    {
-      ...extensions,
-      code,
-    },
-  );
+/*
+ * We didn't track errors addition precisely pre-2.0 and tracking it now has an
+ * unclear ROI, so we just mark all the error code that predates 2.0 as 0.x.
+ */
+const FED1_CODE = '0.x';
+
+export interface ErrorCodeMetadata {
```

Comment:
I kind of disagree here (and I would be against a linter rule that reject this). The error code metadata is used externally, by the doc code generation. Granted, the structural typing of typescript make it so that `ErrorCodeMetadata` is essentially just a name and external code can still describe the type of metadata "manually" without that name if they need to, but if we feel the need to introduce a meaningful name to more conveniently refer to the metadata type, why would we refuse it to others when metadata are exported (and meant to)?

More generally, while I'm all for being careful with what gets exported, I don't think that "only export what is used externally _at the time of the writing_" is a good rule. The whole intend of the approach of this ticket is to make it easier for "external" code to reason about which error code exists and have details on them. Yes it's only used by the doc generation right now, but I could easily see studio eventually have a use for this. And if they do, and do want to refer to the type of metadata in they code, then I could easily see them not bothering opening up a PR to export `ErrorCodeMetadata` then, but instead either redefine it, or inline the type making their code less readable, neither is ideal. To be clear, it's a very minor detail in this case admittedly. I'm just trying to justify why I think it's sometimes ok, and even sometimes a good thing to export thing even if they are not used right away (sometimes, not always, YAGNI is a thing for sure). And I think the exporting is justified here and I'd rather keep it.